src/rust/generator-dispatcher/src/lib.rs

```diff
@@ -121,19 +121,35 @@ impl GeneratorDispatcher {
                                     .await
                                 {
                                     Ok(Some(generator_ids)) => {
-                                        // cache hit
-                                        enqueue_plugin_work(
-                                            plugin_work_queue_client.clone(),
-                                            generator_ids,
-                                            envelope
-                                        ).await?;
+                                        if generator_ids.is_empty() {
+                                            tracing::warn!(
+                                                message = "unrecognized event source",
+                                                tenant_id = %envelope.tenant_id(),
+                                                trace_id = %envelope.trace_id(),
+                                                event_source_id = %envelope.event_source_id(),
```

Comment:
I'd suggest just adding these to a span

Reply:
ok, I'll read up on that

Reply:
I have a plan to do this for every message in the `kafka` lib, will implement in a follow-up PR